exonum-java-binding-core/src/test/java/com/exonum/binding/blockchain/BlockchainIntegrationTest.java

```diff
@@ -0,0 +1,293 @@
+
+import com.exonum.binding.blockchain.Block.Builder;
+import com.exonum.binding.blockchain.serialization.BlockSerializer;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.hash.HashFunction;
+import com.exonum.binding.common.hash.Hashing;
+import com.exonum.binding.common.serialization.StandardSerializers;
+import com.exonum.binding.proxy.Cleaner;
+import com.exonum.binding.proxy.CloseFailuresException;
+import com.exonum.binding.storage.database.Fork;
+import com.exonum.binding.storage.database.MemoryDb;
+import com.exonum.binding.storage.database.Snapshot;
+import com.exonum.binding.storage.indices.ListIndex;
+import com.exonum.binding.storage.indices.ListIndexProxy;
+import com.exonum.binding.storage.indices.MapIndex;
+import com.exonum.binding.storage.indices.MapIndexProxy;
+import com.exonum.binding.storage.indices.ProofListIndexProxy;
+import com.exonum.binding.test.RequiresNativeLibrary;
+import com.exonum.binding.util.LibraryLoader;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@RequiresNativeLibrary
+@ExtendWith(MockitoExtension.class)
+class BlockchainIntegrationTest {
+
+  static {
+    LibraryLoader.load();
+  }
+
+  private static final String TEST_BLOCKS = "test_blocks";
+  private static final String TEST_BLOCK_HASHES = "test_block_hashes";
+  private static final String TEST_BLOCK_TRANSACTIONS = "test_block_transactions";
+
+  @Mock
+  CoreSchemaProxy coreSchema;
+
+  private Blockchain blockchain;
+
+  private MemoryDb database;
+
+  @BeforeEach
+  void setUp() {
+    blockchain = new Blockchain(coreSchema);
+    database = MemoryDb.newInstance();
+  }
+
+  @AfterEach
+  void tearDown() {
+    database.close();
+  }
+
+  @Nested
+  class WithSingleBlock {
+    final List<HashCode> expectedBlockTransactions = ImmutableList.of(HashCode.fromInt(1));
+
+    Cleaner cleaner;
+    Block block;
+
+    @BeforeEach
+    void setUp() {
+      cleaner = new Cleaner();
+      Fork fork = database.createFork(cleaner);
+
+      // Setup blocks
+      MapIndex<HashCode, Block> blocks = MapIndexProxy.newInstance(TEST_BLOCKS, fork,
+          StandardSerializers.hash(), BlockSerializer.INSTANCE);
+      int blockHeight = 1;
+      block = withProperHash(aBlock(blockHeight)
+          .numTransactions(expectedBlockTransactions.size())
+          .build());
+      blocks.put(block.getBlockHash(), block);
+      when(coreSchema.getBlocks()).thenReturn(blocks);
+
+      // Setup tx hashes
```

Comment:
This setup code is needed only for `getBlockTransactions` tests, hence `lenient`. It might be moved in another `@Nested` test class so that it is clear _which_ tests require what initialization code, WDYT about such approach?